features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -35,51 +29,77 @@ xsi:schemaLocation="http://www.osgi.org/xmlns/blueprint/v1.0.0
+        <property name="supportsCategorySearch" value="true"/>
     </bean>
 
-    <bean id="topologyHopProvider" class="org.opennms.features.topology.api.support.VertexHopGraphProvider">
-        <argument ref="topologyProvider"/>
+    <bean id="linkdTopologyProvider" scope="singleton" class="org.opennms.features.topology.plugins.topo.linkd.internal.LinkdTopologyProvider">
+        <argument index="0" value="enlinkd" />
+        <argument index="1" ref="metricRegistry" />
+        <argument index="2"> 
+        	<list>
+        <value>LLDP</value>
+        <value>OSPF</value>
+        <value>ISIS</value>
+        <value>BRIDGE</value>
+        <value>CDP</value>
+        <value>USERDEFINED</value>
+        	</list>
+        	</argument>
+        <argument index="3" ref="onmsTopologyDao"/>
+        <property name="topologyProviderInfo" ref="linkdTopologyProviderInfo"/>
     </bean>
 
-    <!-- Search topology provider -->
-    <bean id="linkdSearchProvider" class="org.opennms.features.topology.plugins.topo.linkd.internal.LinkdSearchProvider">
-        <argument ref="topologyProvider" />
+    <bean id="linkdTopologyHopProvider" class="org.opennms.features.topology.api.support.VertexHopGraphProvider">
+        <argument ref="linkdTopologyProvider"/>
     </bean>
 
-    <service interface="org.opennms.features.topology.api.topo.SearchProvider" ref="linkdSearchProvider" />
-
-    <!-- Meta topology provider -->
     <bean id="linkdMetaTopologyProvider" class="org.opennms.features.topology.api.topo.SimpleMetaTopologyProvider">
-        <argument ref="topologyHopProvider" />
+        <argument ref="linkdTopologyHopProvider" />
     </bean>
-
     <service interface="org.opennms.features.topology.api.topo.MetaTopologyProvider" ref="linkdMetaTopologyProvider">
         <service-properties>
             <entry key="label" value="Enhanced Linkd" />
         </service-properties>
     </service>
 
-    <!-- EnhancedLinkd StatusProviders Service and CheckedOperation -->
-    <bean id="linkdEdgeStatusProvider" class="org.opennms.features.topology.plugins.topo.linkd.internal.LinkdEdgeStatusProvider">
-        <property name="alarmDao" ref="alarmDao"/>
+    <!-- SearchProvider -->
+    <bean id="linkdSearchProvider" class="org.opennms.features.topology.plugins.topo.linkd.internal.LinkdSearchProvider">
+        <argument index="0" ref="linkdTopologyProvider" />
+        <argument index="1" ref="onmsTopologyDao"/>
     </bean>
+    <service interface="org.opennms.features.topology.api.topo.SearchProvider" ref="linkdSearchProvider" />
 
+    <!-- EdgeStatusProvider -->
+    <bean id="linkdEdgeStatusProvider" class="org.opennms.features.topology.plugins.topo.linkd.internal.LinkdEdgeStatusProvider">
+        <argument index="0" ref="alarmDao"/>
+        <argument index="0" ref="onmsTopologyDao"/>
```

Comment:
This is causing the bundle not to start. Index `0` is used twice. 
In general there is no need to use index as it is implicit by the order of the argument, so I'd recommend to omit it in general.

Reply:
I have fixed....
I'm quite busy so I made the changes to the blueprint  without deploiung

Reply:
MArkus, do you have code samples for that?
I know how to do the changes in LinkdTopologyProvider while it takes a List of Protocols Name....so when you select the protocols you just set the protocol list and reload the topology.

Reply:
Why do you not commit the changes? In the branch?